Dockerfile

```diff
@@ -29,7 +29,10 @@ RUN go mod download
 
 COPY . .
 
-RUN make release-binary
+# Propagate Dex version from build args to the build environment
+ARG VERSION
+RUN ([ "$VERSION" != "" ] || export $VERSION) && make release-binary
```

Comment:
I don't think you need to export it. It should be available as an environment variable.

Reply:
Tried this without exporting, and it didn't work 🤷 

It works if I do `ENV VERSION=$VERSION`, but returns an empty version if there is no build arg.

Reply:
Here is a dummy draft that proofs that it should be working: https://github.com/dexidp/dex/pull/3277

If it doesn't, something else is going on.

Can you please check again?

Reply:
Yep, seems like you are right. I tried the most minimalistic script, and it worked. I will revert the `expose` command instructions.